# Análise de Dados - Prouni 2014

Conjunto de dados que apresenta as bolsas concedidas e o perfil dos beneficiários do ProUni de 2014.

Os dados são disponibilizados pelo Site Dados Abertos (http://dadosabertos.mec.gov.br/prouni), em arquivo CSV.

In [1]:
import pandas as pd

df = pd.read_csv ('datasets/pda-prouni-2014.csv', sep=';', encoding = 'ISO-8859-1')

## Conhecendo o dataframe

In [2]:
df.sample(5)

ANO_CONCESSAO_BOLSA  CODIGO_EMEC_IES_BOLSA  \
219174                 2014                   1139   
192113                 2014                   1996   
84224                  2014                   2497   
41992                  2014                   1185   
147946                 2014                    449   

                                           NOME_IES_BOLSA         TIPO_BOLSA  \
219174  FACULDADE DE ESTUDOS ADMINISTRATIVOS DE MINAS ...  BOLSA PARCIAL 50%   
192113               INSTITUTO CUIABÁ DE ENSINO E CULTURA     BOLSA INTEGRAL   
84224                                    FACULDADE ATENEU     BOLSA INTEGRAL   
41992                    CENTRO UNIVERSITÁRIO JORGE AMADO     BOLSA INTEGRAL   
147946                    UNIVERSIDADE LUTERANA DO BRASIL     BOLSA INTEGRAL   

       MODALIDADE_ENSINO_BOLSA       NOME_CURSO_BOLSA NOME_TURNO_CURSO_BOLSA  \
219174              PRESENCIAL                Direito               Matutino   
192113              PRESENCIAL                Direito                Noturno   
84224               PRESENCIAL           Secretariado                Noturno   
41992               PRESENCIAL  Redes De Computadores                Noturno   
147946    EDUCAÇÃO A DISTÂNCIA          Administração      Curso a distância   

       CPF_BENEFICIARIO_BOLSA SEXO_BENEFICIARIO_BOLSA RACA_BENEFICIARIO_BOLSA  \
219174            ***858326**               Masculino                   Parda   
192113            ***875151**               Masculino                   Preta   
84224             ***585813**                Feminino                   Preta   
41992             ***610555**               Masculino                   Parda   
147946            ***254310**               Masculino                   Preta   

       DT_NASCIMENTO_BENEFICIARIO BENEFICIARIO_DEFICIENTE_FISICO  \
219174                 07-10-1994                            NÃO   
192113                 05-12-1989                            NÃO   
84224                  14-03-1996                            NÃO   
41992                  02-01-1985                            NÃO   
147946                 17-04-1971                            NÃO   

       REGIAO_BENEFICIARIO_BOLSA SIGLA_UF_BENEFICIARIO_BOLSA  \
219174                   SUDESTE                          MG   
192113              CENTRO-OESTE                          MT   
84224                   NORDESTE                          CE   
41992                   NORDESTE                          BA   
147946                       SUL                          RS   

       MUNICIPIO_BENEFICIARIO_BOLSA  
219174                       ARINOS  
192113        CHAPADA DOS GUIMARÃES  
84224                     FORTALEZA  
41992                      SALVADOR  
147946                    ENCANTADO

In [3]:
df.shape

(223598, 15)

In [4]:
df.dtypes

ANO_CONCESSAO_BOLSA                int64
CODIGO_EMEC_IES_BOLSA              int64
NOME_IES_BOLSA                    object
TIPO_BOLSA                        object
MODALIDADE_ENSINO_BOLSA           object
NOME_CURSO_BOLSA                  object
NOME_TURNO_CURSO_BOLSA            object
CPF_BENEFICIARIO_BOLSA            object
SEXO_BENEFICIARIO_BOLSA           object
RACA_BENEFICIARIO_BOLSA           object
DT_NASCIMENTO_BENEFICIARIO        object
BENEFICIARIO_DEFICIENTE_FISICO    object
REGIAO_BENEFICIARIO_BOLSA         object
SIGLA_UF_BENEFICIARIO_BOLSA       object
MUNICIPIO_BENEFICIARIO_BOLSA      object
dtype: object

In [5]:
df.isnull().sum()

ANO_CONCESSAO_BOLSA                 0
CODIGO_EMEC_IES_BOLSA               0
NOME_IES_BOLSA                      0
TIPO_BOLSA                          0
MODALIDADE_ENSINO_BOLSA             0
NOME_CURSO_BOLSA                  126
NOME_TURNO_CURSO_BOLSA              0
CPF_BENEFICIARIO_BOLSA              0
SEXO_BENEFICIARIO_BOLSA             0
RACA_BENEFICIARIO_BOLSA             0
DT_NASCIMENTO_BENEFICIARIO          0
BENEFICIARIO_DEFICIENTE_FISICO      0
REGIAO_BENEFICIARIO_BOLSA          21
SIGLA_UF_BENEFICIARIO_BOLSA        21
MUNICIPIO_BENEFICIARIO_BOLSA       21
dtype: int64

#  Dados nulos

- A Coluna NOME_CURSO_BOLSA, consta a falta de 126 registros;


- As Colunas REGIAO_BENEFICIARIO_BOLSA , SIGLA_UF_BENEFICIARIO_BOLSA e MUNICIPIO_BENEFICIARIO_BOLSA, constam falta de 21 registros.

###  Atribuindo valores na Coluna NOME_CURSO_BOLSA

-> Todos os elementos vazios serão preenchidos com o valor definido 'Não Registrado'

In [6]:
df.loc[(df['NOME_CURSO_BOLSA']).isnull()]

ANO_CONCESSAO_BOLSA  CODIGO_EMEC_IES_BOLSA  \
304                    2014                    423   
3377                   2014                    437   
7266                   2014                     82   
8623                   2014                    780   
9087                   2014                    338   
...                     ...                    ...   
218828                 2014                   2271   
218829                 2014                   2271   
218830                 2014                   2271   
219291                 2014                   2271   
219292                 2014                   2271   

                                           NOME_IES_BOLSA      TIPO_BOLSA  \
304     UNIVERSIDADE REGIONAL INTEGRADA DO ALTO URUGUA...  BOLSA INTEGRAL   
3377                              UNIVERSIDADE PARANAENSE  BOLSA INTEGRAL   
7266              UNIVERSIDADE DO OESTE DE SANTA CATARINA  BOLSA INTEGRAL   
8623                               UNIVERSIDADE DE CUIABÁ  BOLSA INTEGRAL   
9087     PONTIFÍCIA UNIVERSIDADE CATÓLICA DE MINAS GERAIS  BOLSA INTEGRAL   
...                                                   ...             ...   
218828                    FACULDADE PITÁGORAS DE IPATINGA  BOLSA INTEGRAL   
218829                    FACULDADE PITÁGORAS DE IPATINGA  BOLSA INTEGRAL   
218830                    FACULDADE PITÁGORAS DE IPATINGA  BOLSA INTEGRAL   
219291                    FACULDADE PITÁGORAS DE IPATINGA  BOLSA INTEGRAL   
219292                    FACULDADE PITÁGORAS DE IPATINGA  BOLSA INTEGRAL   

       MODALIDADE_ENSINO_BOLSA NOME_CURSO_BOLSA NOME_TURNO_CURSO_BOLSA  \
304                 PRESENCIAL              NaN             Vespertino   
3377                PRESENCIAL              NaN               Integral   
7266                PRESENCIAL              NaN                Noturno   
8623                PRESENCIAL              NaN               Matutino   
9087                PRESENCIAL              NaN                Noturno   
...                        ...              ...                    ...   
218828              PRESENCIAL              NaN               Matutino   
218829              PRESENCIAL              NaN               Matutino   
218830              PRESENCIAL              NaN               Matutino   
219291              PRESENCIAL              NaN               Matutino   
219292              PRESENCIAL              NaN               Matutino   

       CPF_BENEFICIARIO_BOLSA SEXO_BENEFICIARIO_BOLSA RACA_BENEFICIARIO_BOLSA  \
304               ***589920**               Masculino                  Branca   
3377              ***239029**                Feminino                  Branca   
7266              ***703229**               Masculino                  Branca   
8623              ***942941**                Feminino                   Parda   
9087              ***160376**                Feminino                  Branca   
...                       ...                     ...                     ...   
218828            ***523466**                Feminino                   Parda   
218829            ***224896**                Feminino                   Parda   
218830            ***529746**                Feminino                  Branca   
219291            ***460356**                Feminino                   Parda   
219292            ***609536**                Feminino                  Branca   

       DT_NASCIMENTO_BENEFICIARIO BENEFICIARIO_DEFICIENTE_FISICO  \
304                    18-01-1994                            NÃO   
3377                   09-05-1993                            NÃO   
7266                   15-10-1990                            NÃO   
8623                   16-11-1986                            NÃO   
9087                   02-09-1994                            NÃO   
...                           ...                            ...   
218828                 24-03-1992                            NÃO   
218829                 24-06

In [7]:
df.update(df['NOME_CURSO_BOLSA'].fillna('Não Registrado'))
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 223598 entries, 0 to 223597
Data columns (total 15 columns):
 #   Column                          Non-Null Count   Dtype 
---  ------                          --------------   ----- 
 0   ANO_CONCESSAO_BOLSA             223598 non-null  int64 
 1   CODIGO_EMEC_IES_BOLSA           223598 non-null  int64 
 2   NOME_IES_BOLSA                  223598 non-null  object
 3   TIPO_BOLSA                      223598 non-null  object
 4   MODALIDADE_ENSINO_BOLSA         223598 non-null  object
 5   NOME_CURSO_BOLSA                223598 non-null  object
 6   NOME_TURNO_CURSO_BOLSA          223598 non-null  object
 7   CPF_BENEFICIARIO_BOLSA          223598 non-null  object
 8   SEXO_BENEFICIARIO_BOLSA         223598 non-null  object
 9   RACA_BENEFICIARIO_BOLSA         223598 non-null  object
 10  DT_NASCIMENTO_BENEFICIARIO      223598 non-null  object
 11  BENEFICIARIO_DEFICIENTE_FISICO  223598 non-null  object
 12  REGIAO_BENEFICIARIO_BOLSA     

### Analisando as Colunas:

#### REGIAO_BENEFICIARIO_BOLSA, SIGLA_UF_BENEFICIARIO_BOLSA e MUNICIPIO_BENEFICIARIO_BOLSA


- 1º: será atribuído o valor 'Não Registrado' nas linhas de cada coluna;


- 2º: Após a análise do df e com o intuito de minimizar esses registros, vê-se a possibilidade de substituição dos valores nas colunas: REGIAO_BENEFICIARIO_BOLSA e SIGLA_UF_BENEFICIARIO_BOLSA

-> essa substituição terá como base os Códigos das IE's ou os Nomes das IE's

-> a substituição ocorrerá somente para as IE's mais frequentes no dataframe


In [8]:
df['REGIAO_BENEFICIARIO_BOLSA'].fillna('Não Registrada', inplace = True)
df['SIGLA_UF_BENEFICIARIO_BOLSA'].fillna('Não Registrada', inplace = True)
df['MUNICIPIO_BENEFICIARIO_BOLSA'].fillna('Não Registrado', inplace = True)

In [9]:
df.loc[(df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada')].head(21)

ANO_CONCESSAO_BOLSA  CODIGO_EMEC_IES_BOLSA  \
47204                  2014                   1493   
48154                  2014                   4086   
76186                  2014                   1806   
79058                  2014                   2150   
80960                  2014                   2773   
85187                  2014                   2773   
86602                  2014                   4452   
86720                  2014                   4452   
93954                  2014                     15   
104318                 2014                  13783   
105433                 2014                   1821   
106231                 2014                   1821   
106232                 2014                   1821   
107046                 2014                   1821   
107047                 2014                   1821   
107048                 2014                   1821   
107049                 2014                   1821   
107050                 2014                   1821   
109528                 2014                   4865   
112286                 2014                   4865   
119823                 2014                   1472   

                                NOME_IES_BOLSA         TIPO_BOLSA  \
47204                    FACULDADE ALVES FARIA     BOLSA INTEGRAL   
48154           FACULDADE DE TECNOLOGIA JARDIM  BOLSA PARCIAL 50%   
76186                    FACULDADE FERNÃO DIAS     BOLSA INTEGRAL   
79058                       FACULDADE DO PIAUÍ     BOLSA INTEGRAL   
80960                      FACULDADE DE MACAPÁ     BOLSA INTEGRAL   
85187                      FACULDADE DE MACAPÁ     BOLSA INTEGRAL   
86602        FACULDADE METROPOLITANA DE MARABÁ  BOLSA PARCIAL 50%   
86720        FACULDADE METROPOLITANA DE MARABÁ     BOLSA INTEGRAL   
93954      UNIVERSIDADE CATÓLICA DE PETRÓPOLIS     BOLSA INTEGRAL   
104318      FACULDADES UNIDAS FEIRA DE SANTANA  BOLSA PARCIAL 50%   
105433                         FACULDADE LIONS     BOLSA INTEGRAL   
106231                         FACULDADE LIONS     BOLSA INTEGRAL   
106232                         FACULDADE LIONS     BOLSA INTEGRAL   
107046                         FACULDADE LIONS     BOLSA INTEGRAL   
107047                         FACULDADE LIONS     BOLSA INTEGRAL   
107048                         FACULDADE LIONS     BOLSA INTEGRAL   
107049                         FACULDADE LIONS     BOLSA INTEGRAL   
107050                         FACULDADE LIONS     BOLSA INTEGRAL   
109528         FACULDADE PITÁGORAS DE SÃO LUIZ     BOLSA INTEGRAL   
112286         FACULDADE PITÁGORAS DE SÃO LUIZ     BOLSA INTEGRAL   
119823  CENTRO UNIVERSITÁRIO LEONARDO DA VINCI     BOLSA INTEGRAL   

       MODALIDADE_ENSINO_BOLSA                               NOME_CURSO_BOLSA  \
47204               PRESENCIAL                                  Administração   
48154               PRESENCIAL                                 Gestão Pública   
76186               PRESENCIAL                             Ciências Contábeis   
79058               PRESENCIAL                                        Direito   
80960               PRESENCIAL                               Gestão Ambiental   
85187               PRESENCIAL                                     Enfermagem   
86602               PRESENCIAL                                   Fisioterapia   
86720               PRESENCIAL                               Engenharia Civil   
93954               PRESENCIAL                               Engenharia Civil   
104318              PRESENCIAL                                    Biomedicina   
105433              PRESENCIAL                                        Direito   
106231              PRESENCIAL  Comunicação Social - Publicidade E Propaganda   
106232              PRESENCIAL  Comunicação Social - Publicidade E Propaganda   
107046              PRESENCIAL                                        Direito   
107047              PRESENCIAL                                        Direito   
1070

In [10]:
#df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE LIONS')].count()
#df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE DE MACAPÁ')].count()
#df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE METROPOLITANA DE MARABÁ')].count()

In [11]:
#df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE LIONS') & (df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada')].count()

In [12]:
#df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE LIONS')].head(23)
#df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE DE MACAPÁ') & (df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada')].head(2)
#df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE METROPOLITANA DE MARABÁ')].sample(10)

### Substituição dos valores nas colunas REGIAO_BENEFICIARIO_BOLSA e SIGLA_UF_BENEFICIARIO_BOLSA

In [13]:
df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE LIONS') & (df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'REGIAO_BENEFICIARIO_BOLSA'] = 'CENTRO-OESTE'
df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE DE MACAPÁ') & (df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'REGIAO_BENEFICIARIO_BOLSA'] = 'NORTE'
df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE METROPOLITANA DE MARABÁ') & (df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'REGIAO_BENEFICIARIO_BOLSA'] = 'NORTE'

In [14]:
df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE LIONS') & (df['SIGLA_UF_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'SIGLA_UF_BENEFICIARIO_BOLSA'] = 'GO'
df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE DE MACAPÁ') & (df['SIGLA_UF_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'SIGLA_UF_BENEFICIARIO_BOLSA'] = 'AP'
df.loc[(df['NOME_IES_BOLSA'] == 'FACULDADE METROPOLITANA DE MARABÁ') & (df['SIGLA_UF_BENEFICIARIO_BOLSA'] == 'Não Registrada'), 'SIGLA_UF_BENEFICIARIO_BOLSA'] = 'PA'

In [15]:
df.loc[(df['REGIAO_BENEFICIARIO_BOLSA'] == 'Não Registrada')].count()

ANO_CONCESSAO_BOLSA               9
CODIGO_EMEC_IES_BOLSA             9
NOME_IES_BOLSA                    9
TIPO_BOLSA                        9
MODALIDADE_ENSINO_BOLSA           9
NOME_CURSO_BOLSA                  9
NOME_TURNO_CURSO_BOLSA            9
CPF_BENEFICIARIO_BOLSA            9
SEXO_BENEFICIARIO_BOLSA           9
RACA_BENEFICIARIO_BOLSA           9
DT_NASCIMENTO_BENEFICIARIO        9
BENEFICIARIO_DEFICIENTE_FISICO    9
REGIAO_BENEFICIARIO_BOLSA         9
SIGLA_UF_BENEFICIARIO_BOLSA       9
MUNICIPIO_BENEFICIARIO_BOLSA      9
dtype: int64

### Excluindo as Colunas CPF_BENEFICIARIO_BOLSA e MUNICIPIO_BENEFICIARIO_BOLSA

In [16]:
df.drop(['CPF_BENEFICIARIO_BOLSA', 'MUNICIPIO_BENEFICIARIO_BOLSA'], axis = 1, inplace = True)

### Renomeando Colunas

In [17]:
df.columns

Index(['ANO_CONCESSAO_BOLSA', 'CODIGO_EMEC_IES_BOLSA', 'NOME_IES_BOLSA',
       'TIPO_BOLSA', 'MODALIDADE_ENSINO_BOLSA', 'NOME_CURSO_BOLSA',
       'NOME_TURNO_CURSO_BOLSA', 'SEXO_BENEFICIARIO_BOLSA',
       'RACA_BENEFICIARIO_BOLSA', 'DT_NASCIMENTO_BENEFICIARIO',
       'BENEFICIARIO_DEFICIENTE_FISICO', 'REGIAO_BENEFICIARIO_BOLSA',
       'SIGLA_UF_BENEFICIARIO_BOLSA'],
      dtype='object')

In [18]:
df.rename(columns = {'ANO_CONCESSAO_BOLSA':'ANO_CONCESSAO', 
                     'CODIGO_EMEC_IES_BOLSA':'CODIGO_IES', 
                     'NOME_IES_BOLSA':'NOME_IES', 
                     'MODALIDADE_ENSINO_BOLSA':'MODALIDADE_ENSINO', 
                     'NOME_CURSO_BOLSA':'NOME_CURSO'}, inplace = True)

In [19]:
df.rename(columns = {'NOME_TURNO_CURSO_BOLSA':'TURNO_CURSO', 
                     'SEXO_BENEFICIARIO_BOLSA':'SEXO_BENEFICIARIO', 
                     'RACA_BENEFICIARIO_BOLSA':'RACA_BENEFICIARIO', 
                     'REGIAO_BENEFICIARIO_BOLSA':'REGIAO_BENEFICIARIO', 
                     'SIGLA_UF_BENEFICIARIO_BOLSA':'UF_BENEFICIARIO', 
                     'DT_NASCIMENTO_BENEFICIARIO':'ANO_NASCIMENTO_BENEFICIARIO'}, inplace = True)

### Descobrindo a idade aproximada* dos beneficiários

* é aproximada, pois o dataframe não menciona qual semestre do ano o beneficiário foi contemplado pela bolsa.

-> Criar uma nova coluna IDADE_APROX_BENEFICIARIO para calcular a idade aproximada deles

-> Alterar o formato da coluna DT_NASCIMENTO_BENEFICIARIO Para Y

-> Calcular com base na coluna ANO_CONCESSAO a idade aproximada do beneficiário

In [20]:
df['ANO_NASCIMENTO_BENEFICIARIO'] = pd.to_datetime(df['ANO_NASCIMENTO_BENEFICIARIO']).dt.year
print(df['ANO_NASCIMENTO_BENEFICIARIO'])

0         1978
1         1966
2         1996
3         1993
4         1990
          ... 
223593    1982
223594    1987
223595    1993
223596    1992
223597    1974
Name: ANO_NASCIMENTO_BENEFICIARIO, Length: 223598, dtype: int64


In [21]:
df.insert(loc=10, column='IDADE_APROX_BENEFICIARIO', value=10)

In [22]:
df.head(1)

ANO_CONCESSAO  CODIGO_IES                     NOME_IES         TIPO_BOLSA  \
0           2014          20  UNIVERSIDADE DE PASSO FUNDO  BOLSA PARCIAL 50%   

  MODALIDADE_ENSINO     NOME_CURSO TURNO_CURSO SEXO_BENEFICIARIO  \
0        PRESENCIAL  Administração     Noturno         Masculino   

  RACA_BENEFICIARIO  ANO_NASCIMENTO_BENEFICIARIO  IDADE_APROX_BENEFICIARIO  \
0            Branca                         1978                        10   

  BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO UF_BENEFICIARIO  
0                            NÃO                 SUL              RS

In [23]:
df['IDADE_APROX_BENEFICIARIO'] = df['ANO_CONCESSAO'] - df['ANO_NASCIMENTO_BENEFICIARIO']
df.sample(5)

ANO_CONCESSAO  CODIGO_IES  \
15763            2014         298   
189574           2014        2754   
152304           2014          82   
117507           2014        1107   
1597             2014        1032   

                                           NOME_IES      TIPO_BOLSA  \
15763                 UNIVERSIDADE PITÁGORAS UNOPAR  BOLSA INTEGRAL   
189574                          FACULDADE SÃO PAULO  BOLSA INTEGRAL   
152304      UNIVERSIDADE DO OESTE DE SANTA CATARINA  BOLSA INTEGRAL   
117507        Centro Universitário Estácio do Ceará  BOLSA INTEGRAL   
1597    CENTRO UNIVERSITÁRIO SALESIANO DE SÃO PAULO  BOLSA INTEGRAL   

           MODALIDADE_ENSINO                          NOME_CURSO  \
15763   EDUCAÇÃO A DISTÂNCIA                    Gestão Ambiental   
189574            PRESENCIAL                          Enfermagem   
152304            PRESENCIAL                           Agronomia   
117507            PRESENCIAL                  Ciências Contábeis   
1597              PRESENCIAL  Engenharia De Automação E Controle   

              TURNO_CURSO SEXO_BENEFICIARIO RACA_BENEFICIARIO  \
15763   Curso a distância          Feminino             Parda   
189574            Noturno          Feminino             Parda   
152304            Noturno         Masculino             Parda   
117507         Vespertino          Feminino             Preta   
1597              Noturno         Masculino            Branca   

        ANO_NASCIMENTO_BENEFICIARIO  IDADE_APROX_BENEFICIARIO  \
15763                          1974                        40   
189574                         1993                        21   
152304                         1994                        20   
117507                         1991                        23   
1597                           1992                        22   

       BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO UF_BENEFICIARIO  
15763                             NÃO            NORDESTE              BA  
189574                            NÃO               NORTE              RO  
152304                            NÃO                 SUL              SC  
117507                            NÃO            NORDESTE              CE  
1597                              NÃO             SUDESTE              SP

### Criando intervalos em que se agrupam pessoas de idade aproximada

-> Criar uma nova coluna FAIXA_ETA_BENEFICIARIO

-> Popular a coluna seguindo os seguintes critérios:

    Menos de 18 anos
    De 18 a 24 anos
    De 24 a 30 anos
    De 30 a 36 anos
    De 36 a 48 anos
    De 48 a 60 anos
    Mais de 60 anos

In [24]:
df.insert(loc=11, column='FAIXA_ETA_BENEFICIARIO', value='Idade')

In [25]:
df.head()

ANO_CONCESSAO  CODIGO_IES  \
0           2014          20   
1           2014          20   
2           2014         423   
3           2014         423   
4           2014         423   

                                            NOME_IES         TIPO_BOLSA  \
0                        UNIVERSIDADE DE PASSO FUNDO  BOLSA PARCIAL 50%   
1                        UNIVERSIDADE DE PASSO FUNDO  BOLSA PARCIAL 50%   
2  UNIVERSIDADE REGIONAL INTEGRADA DO ALTO URUGUA...     BOLSA INTEGRAL   
3  UNIVERSIDADE REGIONAL INTEGRADA DO ALTO URUGUA...  BOLSA PARCIAL 50%   
4  UNIVERSIDADE REGIONAL INTEGRADA DO ALTO URUGUA...     BOLSA INTEGRAL   

  MODALIDADE_ENSINO           NOME_CURSO TURNO_CURSO SEXO_BENEFICIARIO  \
0        PRESENCIAL        Administração     Noturno         Masculino   
1        PRESENCIAL        Administração     Noturno         Masculino   
2        PRESENCIAL  Ciências Biológicas     Noturno          Feminino   
3        PRESENCIAL  Ciências Biológicas     Noturno          Feminino   
4        PRESENCIAL            Agronomia     Noturno          Feminino   

  RACA_BENEFICIARIO  ANO_NASCIMENTO_BENEFICIARIO  IDADE_APROX_BENEFICIARIO  \
0            Branca                         1978                        36   
1            Branca                         1966                        48   
2            Branca                         1996                        18   
3            Branca                         1993                        21   
4            Branca                         1990                        24   

  FAIXA_ETA_BENEFICIARIO BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO  \
0                  Idade                            NÃO                 SUL   
1                  Idade                            NÃO                 SUL   
2                  Idade                            NÃO                 SUL   
3                  Idade                            NÃO                 SUL   
4                  Idade                            NÃO                 SUL   

  UF_BENEFICIARIO  
0              RS  
1              RS  
2              RS  
3              RS  
4              RS

In [26]:
for i in range(len (df)): 
    if (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] > 0) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 18):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'Menos de 18 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 18) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 24):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 18 a 24 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 24) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 30):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 24 a 30 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 30) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 36):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 30 a 36 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 36) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 48):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 36 a 48 anos'
    
    elif (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] >= 48) & (df.loc[i, 'IDADE_APROX_BENEFICIARIO'] < 60):
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'De 48 a 60 anos'
    
    else:
        df.loc[i, 'FAIXA_ETA_BENEFICIARIO'] = 'Mais de 60 anos'

In [27]:
df.sample()

ANO_CONCESSAO  CODIGO_IES                          NOME_IES  \
190143           2014        1711  FACULDADE DE PRESIDENTE PRUDENTE   

            TIPO_BOLSA MODALIDADE_ENSINO          NOME_CURSO TURNO_CURSO  \
190143  BOLSA INTEGRAL        PRESENCIAL  Ciências Contábeis     Noturno   

       SEXO_BENEFICIARIO RACA_BENEFICIARIO  ANO_NASCIMENTO_BENEFICIARIO  \
190143          Feminino             Parda                         1989   

        IDADE_APROX_BENEFICIARIO FAIXA_ETA_BENEFICIARIO  \
190143                        25        De 24 a 30 anos   

       BENEFICIARIO_DEFICIENTE_FISICO REGIAO_BENEFICIARIO UF_BENEFICIARIO  
190143                            NÃO             SUDESTE              SP

# Analisando os dados

### Tipos de bolsas

In [28]:
df['TIPO_BOLSA'].value_counts()

BOLSA INTEGRAL       176033
BOLSA PARCIAL 50%     47565
Name: TIPO_BOLSA, dtype: int64

### Modalidade de ensino

In [29]:
df['MODALIDADE_ENSINO'].value_counts()

PRESENCIAL              177536
EDUCAÇÃO A DISTÂNCIA     46062
Name: MODALIDADE_ENSINO, dtype: int64

### Top 10 - Cursos

In [30]:
df['NOME_CURSO'].value_counts().head(10)

Administração                 27163
Direito                       18712
Pedagogia                     16337
Ciências Contábeis            13638
Gestão De Recursos Humanos     8807
Engenharia Civil               8498
Enfermagem                     7769
Educação Física                6668
Serviço Social                 5974
Psicologia                     5590
Name: NOME_CURSO, dtype: int64

### Turno predominante

In [31]:
df['TURNO_CURSO'].value_counts()

Noturno              127635
Curso a distância     46062
Matutino              38254
Integral               6247
Vespertino             5400
Name: TURNO_CURSO, dtype: int64

### Sexo dos beneficiários

In [32]:
df['SEXO_BENEFICIARIO'].value_counts()

Feminino     126837
Masculino     96761
Name: SEXO_BENEFICIARIO, dtype: int64

### Raça dos beneficiários

In [33]:
df['RACA_BENEFICIARIO'].value_counts()

Parda            96935
Branca           93180
Preta            29291
Amarela           3841
Indígena           181
Não Informada      170
Name: RACA_BENEFICIARIO, dtype: int64

### Faixa etária dos Beneficiários

In [34]:
df['FAIXA_ETA_BENEFICIARIO'].value_counts()

De 18 a 24 anos     131183
De 24 a 30 anos      45897
De 30 a 36 anos      23079
De 36 a 48 anos      17420
De 48 a 60 anos       3229
Menos de 18 anos      2580
Mais de 60 anos        210
Name: FAIXA_ETA_BENEFICIARIO, dtype: int64

### Beneficiário deficientes físicos

In [35]:
df['BENEFICIARIO_DEFICIENTE_FISICO'].value_counts()

NÃO    221825
SIM      1773
Name: BENEFICIARIO_DEFICIENTE_FISICO, dtype: int64

### Região x Bolsas

In [36]:
df['REGIAO_BENEFICIARIO'].value_counts()

SUDESTE           106137
SUL                41579
NORDESTE           39535
CENTRO-OESTE       21672
NORTE              14666
Não Registrada         9
Name: REGIAO_BENEFICIARIO, dtype: int64

### Top 5 -  UF's x Bolsas

In [37]:
df['UF_BENEFICIARIO'].value_counts().head()

SP    62029
MG    26827
RS    19252
PR    14715
RJ    13017
Name: UF_BENEFICIARIO, dtype: int64

### Top 10 - IE's x Bolsas

In [38]:
df['NOME_IES'].value_counts().head(10)

UNIVERSIDADE PITÁGORAS UNOPAR                       17443
UNIVERSIDADE PAULISTA                               10414
UNIVERSIDADE ANHANGUERA - UNIDERP                    7634
UNIVERSIDADE ESTÁCIO DE SÁ                           6706
UNIVERSIDADE SÃO JUDAS TADEU                         4264
UNIVERSIDADE LUTERANA DO BRASIL                      4206
UNIVERSIDADE NOVE DE JULHO                           3394
PONTIFÍCIA UNIVERSIDADE CATÓLICA DE MINAS GERAIS     3298
UNIVERSIDADE DO OESTE PAULISTA                       2807
CENTRO UNIVERSITÁRIO LEONARDO DA VINCI               2717
Name: NOME_IES, dtype: int64

## Criando um novo arquivo csv

In [39]:
novo_df = df

In [41]:
novo_df.to_csv('N_Datasets/Novo_Prouni2014.csv', sep=';', encoding = 'ISO-8859-1', index = False)